Machine stats guide (#117)
tidalmigrations/tidalmigrations.github.io: pages/tidal tools/machine_stats.md

of each application's resource requirements in the cloud and ignores the
trickery of thin-provisioning from SAN tools.  This allows you to confidently
plan data replication time, or other migration metrics on an app-by-app basis.

{% include tip.html content="As your cloud migration will likely take place
over many months or years, it's important to have current resource requirements
throughout your journey. To accomplish this, setup `machine_stats` in a cron
job or Scheduled Task and synchronize your data on a _daily_ basis to Tidal
Migrations." %}

## Windows

### Introduction

This is a simple and effective way to gather machine stats (RAM, Storage,
CPU) from a Windows Server environment. It uses
[WinRM](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management)
to `Invoke-Command` across your servers, creating a JSON file to securely send
to your Tidal Migrations instance using the tidal command.

### Requirements and Dependencies

Prior to the steps on syncing your hypervisors, be sure to have the below
dependencies setup:

- To get started you will need to have Tidal Tools installed. You can check out

>   this [article](tidal-tools.html) on how to install it.

Reviewer: QQ, is it an article or a guide? I guess something times we use those words interchangeably. For me, `guide` sounds more appropriate. What do you think?

Author: Good catch! Yes, let's stick with “guide”!

Reviewer: FYI, we have the same on line 94

Migrations." %}

## Windows

### Introduction

This is a simple and effective way to gather machine stats (RAM, Storage,
CPU) from a Windows Server environment. It uses
[WinRM](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management)
to `Invoke-Command` across your servers, creating a JSON file to securely send
to your Tidal Migrations instance using the tidal command.

### Requirements and Dependencies

Prior to the steps on syncing your hypervisors, be sure to have the below
dependencies setup:

- To get started you will need to have Tidal Tools installed. You can check out
  [Getting Started with Tidal Tools](tidal-tools.html) guide on how to install
  it.
- You will need [WinRM
  enabled](https://support.auvik.com/hc/en-us/articles/204424994-How-to-enable-WinRM-with-domain-controller-Group-Policy-for-WMI-monitoring)
  across your environment for this.
- The scripts needed for this process can be found in [Machine Stats for
  Windows GitHub
  repository](https://github.com/tidalmigrations/machine_stats/tree/master/windows).

### Running the Script

> Start syncing your Hyper-V with Tidal Migrations, by running the script:

Reviewer 2: ```suggestion
Start syncing your servers with Tidal Migrations, by running the script:
```

`runner.ps1`.

{% include note.html content="Be sure to set PowerShell's Execution policy to
execute this script before running it. [This
guide](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-6)
will help you in doing so." %}

1. It will prompt you to set your username.

2. Then, run the script: `save_password.ps1` and enter your password when
   prompted. This will save your credentials securely in a text file.

3. Save a list of the server hostnames that you would like to sync in a file
   called: _servers.txt_.  The data of these servers will be shown in a table,
   with numerical values rounded to the nearest second decimal.

{% include note.html content="We recommend that you set this script to run

> periodically so that your Hyper-V servers are synced on a daily basis and the

Reviewer 2: ```suggestion
periodically so that your servers are synced on a daily basis and the
```

data is up to date. Here is an [example](
https://docs.microsoft.com/en-us/windows/desktop/taskschd/daily-trigger-example--scripting-)
of a scripting task in PowerShell that runs daily."%}

> And there you have it! Your Hyper-V servers will be synced to Tidal Migrations.

Reviewer 2: ```suggestion
And there you have it! Your servers will be synced to Tidal Migrations.
```
